src/routes/evalutation/index.ts

```diff
@@ -0,0 +1,147 @@
+import { OpenAPIHono } from "@hono/zod-openapi";
+import prisma from "../../lib/prisma-client.js";
+import {
+  createEvaluation,
+  getEvaluation,
+  getEvaluationById,
+  deleteEvaluation,
+} from "./route.js";
+import { checkRole } from "../../lib/auth-provider.js";
+import { Role } from "@prisma/client";
+
+const evaluationRouter = new OpenAPIHono();
+
+interface JWTPayload {
+  userId: string;
+  role: Role;
+  exp: number;
+}
+declare module "hono" {
+  interface ContextVariableMap {
+    user: JWTPayload;
+  }
+}
+
+// PUT /evaluations - Entering the scores for a project
+evaluationRouter.openapi(createEvaluation, async (ctx) => {
+  try {
+    const allowedRoles: Role[] = ["SUPER_ADMIN", "EVALUATOR"];
+    const { projectId, score } = ctx.req.valid("json");
+    if (!checkRole(allowedRoles, ctx)) {
+      return ctx.text("You do not have permission to create evaluations", 403);
+    }
+
+    if (score < 0 || score > 10) {
+      return ctx.text("Score must be between 0 and 10", 400);
+    }
+    await prisma.evaluation.updateMany({
+      where: { projectId },
+      data: { score },
+    });
+
+    return ctx.json("Record updated", 201);
+  } catch (error) {
+    console.error("Error occurred:", error);
+
+    //Foreign key constraint error
+    if ((error as { code?: string }).code === "P2003") {
+      return ctx.text("Invalid project ID - Project does not exist", 400);
+    }
+    return ctx.text("An unexpected error occurred", 500);
+  }
+});
+
+// GET /evaluations - Get all evaluations
+evaluationRouter.openapi(getEvaluation, async (ctx) => {
+  try {
+    const allowedRoles: Role[] = ["SUPER_ADMIN", "EVALUATOR"];
+    if (!checkRole(allowedRoles, ctx)) {
+      return ctx.text("You do not have permission to view evaluations", 403);
+    } else {
+      const evaluations = await prisma.evaluation.findMany();
+      return ctx.json(evaluations, 200);
+    }
+  } catch (error) {
+    console.error("Error occurred:", error);
+    return ctx.text("An unexpected error occurred", 500);
+  }
+});
+
+// GET /evaluations/:id - Get a specific evaluation by projectId
+evaluationRouter.openapi(getEvaluationById, async (ctx) => {
+  const projectId = ctx.req.param("id");
+  try {
+    const user = ctx.get("user");
```

Comment:
replace with `const user = getCurrentUser(ctx);` from `auth-provider` it will fix the jwt type error.